### ISSUE TITLE
mix surface.init: no function clause matching in Sourceror.Zipper.down/1

### ISSUE THREAD

**Issue by kopchik** [surface-ui/surface#707]
## Describe the bug

`mix surface.init` fails on patching. I tested both on existing project, and also on a fresh project.

## How to reproduce it

I just do `mix surface.init --demo --catalogue --layouts --yes` on my project, and it fails with:
```
* patching lib/alex_web.ex
** (FunctionClauseError) no function clause matching in Sourceror.Zipper.down/1    
    
    The following arguments were given to Sourceror.Zipper.down/1:
    
        # 1
        nil
    
    Attempted function clauses (showing 1 out of 1):
    
        def down(%Sourceror.Zipper{node: tree} = zipper)
    
    (sourceror 0.13.0) lib/sourceror/zipper.ex:107: Sourceror.Zipper.down/1
    (surface 0.11.0) lib/mix/tasks/surface/surface.init/ex_patcher_move.ex:129: Mix.Tasks.Surface.Init.ExPatcher.Move.last_arg/1
    (surface 0.11.0) lib/mix/tasks/surface/surface.init/ex_patcher_move.ex:135: Mix.Tasks.Surface.Init.ExPatcher.Move.body/1
    (surface 0.11.0) lib/mix/tasks/surface/surface.init/ex_patcher.ex:82: Mix.Tasks.Surface.Init.ExPatcher.move/2
    (surface 0.11.0) lib/mix/tasks/surface/surface.init/file_patchers/phoenix.ex:158: Mix.Tasks.Surface.Init.FilePatchers.Phoenix.append_def_to_web_module/3
    (surface 0.11.0) lib/mix/tasks/surface/surface.init/patcher.ex:113: Mix.Tasks.Surface.Init.Patcher.run_patch_funs/4
    (surface 0.11.0) lib/mix/tasks/surface/surface.init/patcher.ex:28: anonymous fn/3 in Mix.Tasks.Surface.Init.Patcher.patch_file/3
    (elixir 1.15.4) lib/enum.ex:2510: Enum."-reduce/3-lists^foldl/2-0-"/3
    (surface 0.11.0) lib/mix/tasks/surface/surface.init/patcher.ex:27: anonymous fn/3 in Mix.Tasks.Surface.Init.Patcher.patch_file/3
    (surface 0.11.0) lib/mix/tasks/surface/surface.init/patcher.ex:148: Mix.Tasks.Surface.Init.Patcher.log/3
    (elixir 1.15.4) lib/enum.ex:1693: Enum."-map/2-lists^map/1-1-"/2
    (elixir 1.15.4) lib/enum.ex:1693: Enum."-map/2-lists^map/1-1-"/2
    (surface 0.11.0) lib/mix/tasks/surface/surface.init/project_patcher.ex:19: Mix.Tasks.Surface.Init.ProjectPatcher.run/2
    (surface 0.11.0) lib/mix/tasks/surface/surface.init.ex:124: Mix.Tasks.Surface.Init.run/1
    (mix 1.15.4) lib/mix/task.ex:447: anonymous fn/3 in Mix.Task.run_task/5
    (mix 1.15.4) lib/mix/cli.ex:92: Mix.CLI.run_task/2
    /usr/local/bin/mix:2: (file)
    (elixir 1.15.4) lib/code.ex:1435: Code.require_file/2
```

Content of `alex_web.ex`:
```elixir
defmodule AlexWeb do
  @moduledoc """
  The entrypoint for defining your web interface, such
  as controllers, components, channels, and so on.

  This can be used in your application as:

      use AlexWeb, :controller
      use AlexWeb, :html

  The definitions below will be executed for every controller,
  component, etc, so keep them short and clean, focused
  on imports, uses and aliases.

  Do NOT define functions inside the quoted expressions
  below. Instead, define additional modules and import
  those modules here.
  """

  def static_paths, do: ~w(assets fonts images favicon.ico robots.txt)

  def router do
    quote do
      use Phoenix.Router, helpers: false

      # Import common connection and controller functions to use in pipelines
      import Plug.Conn
      import Phoenix.Controller
      import Phoenix.LiveView.Router
    end
  end

  def channel do
    quote do
      use Phoenix.Channel
    end
  end

  def controller do
    quote do
      use Phoenix.Controller,
        formats: [:html, :json],
        layouts: [html: AlexWeb.Layouts]

      import Plug.Conn
      import AlexWeb.Gettext

      unquote(verified_routes())
    end
  end

  def live_view do
    quote do
      use Phoenix.LiveView,
        layout: {AlexWeb.Layouts, :app}

      unquote(html_helpers())
    end
  end

  def live_component do
    quote do
      use Phoenix.LiveComponent

      unquote(html_helpers())
    end
  end

  def html do
    quote do
      use Phoenix.Component

      # Import convenience functions from controllers
      import Phoenix.Controller,
        only: [get_csrf_token: 0, view_module: 1, view_template: 1]

      # Include general helpers for rendering HTML
      unquote(html_helpers())
    end
  end

  defp html_helpers do
    quote do
      # HTML escaping functionality
      import Phoenix.HTML
      # Core UI components and translation
      import AlexWeb.CoreComponents
      import AlexWeb.Gettext

      # Shortcut for generating JS commands
      alias Phoenix.LiveView.JS

      # Routes generation with the ~p sigil
      unquote(verified_routes())
    end
  end

  def verified_routes do
    quote do
      use Phoenix.VerifiedRoutes,
        endpoint: AlexWeb.Endpoint,
        router: AlexWeb.Router,
        statics: AlexWeb.static_paths()
    end
  end

  @doc """
  When used, dispatch to the appropriate controller/view/etc.
  """
  defmacro __using__(which) when is_atom(which) do
    apply(__MODULE__, which, [])
  end
end
```

## The behavior you expected

I'd expect not to see a backtrace.

## Your Environment
Surface: v0.11.0
LiveView: v0.19.5
Elixir: v1.15.4

PS can also be reproduced like this:
```sh
mix phx.new my_app
cd my_app/
vim mix.exs  # add surface to dependencies
mix deps.get
mix surface.init --demo --catalogue --layouts
```

**Comment by Nialisc** [comment]
Same issue here :/

**Comment by msaraiva** [comment]
Hi @kopchik! 

We have a quick fix for this in https://github.com/surface-ui/surface/commit/aae0c0690d535cbe2760f003a544c8eff337bbc4. A permanent solution will come soon as we adjust the code to apply to the changes in the `sourceror` lib.

You can also work around this issue in your current surface version by explicitly adding `{:sourceror, "~> 0.12.0"},` to your project's `mix.exs`.

Thanks for reporting this.

**Comment by Haseenah01** [comment]
> Hi @kopchik!
> 
> We have a quick fix for this in [aae0c06](https://github.com/surface-ui/surface/commit/aae0c0690d535cbe2760f003a544c8eff337bbc4). A permanent solution will come soon as we adjust the code to apply to the changes in the `sourceror` lib.
> 
> You can also work around this issue in your current surface version by explicitly adding `{:sourceror, "~> 0.12.0"},` to your project's `mix.exs`.
> 
> Thanks for reporting this.

Hi, I have also experienced the same problem, I added the sourceror dependency in my mix.exs file as well, however the issue still persists. Any idea on how to fix it?
